OrbitClientModel/CaptureDeserializerTest.cpp

```diff
@@ -0,0 +1,408 @@
+  header.set_version(capture_deserializer::internal::kRequiredCaptureVersion);
+
+  std::string serialized_header;
+  header.SerializeToString(&serialized_header);
+  int32_t size_of_header = serialized_header.size();
+  std::stringstream stream;
+  stream << std::string(absl::bit_cast<char*>(&size_of_header), sizeof(size_of_header))
+         << serialized_header;
+
+  capture_deserializer::Load(stream, "not_existing_test_file", &listener, &cancellation_requested);
+}
+
+TEST(CaptureDeserializer, LoadCaptureInfoOnCaptureStarted) {
+  MockCaptureListener listener;
+  CaptureInfo capture_info;
+  capture_info.set_process_id(42);
+  capture_info.set_process_name("process");
+  FunctionInfo* selected_function = capture_info.add_selected_functions();
+  selected_function->set_name("foo");
+  selected_function->set_pretty_name("void foo()");
+  selected_function->set_address(21);
+  selected_function->set_size(12);
+  std::atomic<bool> cancellation_requested = false;
+  uint8_t empty_data = 0;
+  google::protobuf::io::CodedInputStream empty_stream(&empty_data, 0);
+
+  absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionInfo> actual_selected_functions;
+  // There will be no call to OnCaptureStarted other then the one specified next.
+  EXPECT_CALL(listener, OnCaptureStarted(_, _, _, _, _)).Times(0);
+  EXPECT_CALL(listener, OnCaptureStarted(42, "process", NotNull(), _, IsEmpty()))
+      .Times(1)
+      .WillOnce(SaveArg<3>(&actual_selected_functions));
+  EXPECT_CALL(listener, OnCaptureComplete).Times(1);
+  EXPECT_CALL(listener, OnCaptureFailed).Times(0);
+  EXPECT_CALL(listener, OnCaptureCancelled).Times(0);
+
+  EXPECT_CALL(listener, OnAddressInfo).Times(0);
+  EXPECT_CALL(listener, OnThreadName).Times(0);
+
+  capture_deserializer::internal::LoadCaptureInfo(capture_info, &listener, &empty_stream,
+                                                  &cancellation_requested);
+
+  ASSERT_EQ(actual_selected_functions.size(), 1);
+  ASSERT_TRUE(actual_selected_functions.contains(selected_function->address()));
+  FunctionInfo actual_function_info = actual_selected_functions.at(selected_function->address());
+
+  EXPECT_EQ(actual_function_info.name(), selected_function->name());
+  EXPECT_EQ(actual_function_info.pretty_name(), selected_function->pretty_name());
+  EXPECT_EQ(actual_function_info.size(), selected_function->size());
+}
+
+TEST(CaptureDeserializer, LoadCaptureInfoAddressInfos) {
+  MockCaptureListener listener;
+  CaptureInfo capture_info;
+
+  LinuxAddressInfo address_info_1;
+  address_info_1.set_function_name("foo");
+  address_info_1.set_module_path("/path");
+  address_info_1.set_offset_in_function(0);
+  address_info_1.set_absolute_address(123);
+  capture_info.add_address_infos()->CopyFrom(address_info_1);
+
+  LinuxAddressInfo address_info_2;
+  address_info_2.set_function_name("bar");
+  address_info_2.set_module_path("/path");
+  address_info_2.set_offset_in_function(0);
+  address_info_2.set_absolute_address(123);
```

Comment:
Shouldn't be data in `address_info_2` different from `address_info_1`?

Reply:
Sure, thats better